plugins/performance-lab/includes/site-health/audit-enqueued-assets/helper.php

```diff
@@ -279,31 +279,50 @@ function perflab_aea_get_total_size_bytes_enqueued_styles() {
 }
 
 /**
- * Convert full URL paths to absolute paths.
- * Covers Standard WP configuration, wp-content outside WP directories and subdirectories.
- * Ex: https://example.com/content/themes/, https://example.com/wp/wp-includes/
+ * Gets the size of the asset in bytes.
  *
- * @since 1.0.0
+ * @since n.e.x.t
  *
- * @param string $resource_url URl resource link.
- * @return string Returns absolute path to the resource.
+ * @param string $resource_url URL of the resource.
+ * @return int|null Returns the size in bytes of the asset, or null if it cannot be determined.
  */
-function perflab_aea_get_path_from_resource_url( string $resource_url ): string {
-	if ( '' === $resource_url ) {
-		return '';
-	}
+function perflab_aea_get_asset_size( string $resource_url ): ?int {
+	$response = wp_remote_get(
+		$resource_url,
+		array(
+			'timeout' => 10,
+			'headers' => perflab_aea_copy_basic_auth_headers( array() ),
+		)
+	);
 
-	// Different content folder ex. /content/.
-	if ( 0 === strpos( $resource_url, content_url() ) ) {
-		return WP_CONTENT_DIR . substr( $resource_url, strlen( content_url() ) );
+	if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+		return null;
 	}
 
-	// wp-content in a subdirectory. ex. /blog/wp-content/.
-	$site_url = untrailingslashit( site_url() );
-	if ( 0 === strpos( $resource_url, $site_url ) ) {
-		return untrailingslashit( ABSPATH ) . substr( $resource_url, strlen( $site_url ) );
+	$body = wp_remote_retrieve_body( $response );
+	if ( '' === $body ) {
+		return null;
```

Comment:
Could it be that the body length is actually zero in this case?

Reply:
I’m not understanding what you’re trying to suggest. If you’re referring to cases when an asset has no content and the server returns an empty string, this will ignore such assets. Should these assets still be included in the asset count?

Reply:
Yes, but the issue is that a zero-size body would still be a blocking request. 

So I think this condition should be removed.

```suggestion
	$body = wp_remote_retrieve_body( $response );
```

Reply:
Done in https://github.com/WordPress/performance/pull/2059/commits/0be6561da09ef9d04f66c8ed99d3daeb9ec38308.